lib/CMakeLists.txt

```diff
@@ -64,27 +70,51 @@ if (SWIFT_SWIFT_PARSER)
-      DEPENDS "${SWIFT_SYNTAX_LIBRARIES_BUILD_DIR}/${sharedlib}"
-      COMMAND ${CMAKE_COMMAND} -E copy_if_different ${SWIFT_SYNTAX_LIBRARIES_BUILD_DIR}/${sharedlib} ${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}
-      ${add_origin_rpath}
-    )
-
-    add_custom_target(copy_swiftSyntaxLibrary_${sharedlib}
-      DEPENDS "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}"
-      COMMENT "Copying ${sharedlib}"
-    )
-
-    swift_install_in_component(
-      PROGRAMS "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}"
-      DESTINATION "lib${LLVM_LIBDIR_SUFFIX}/swift/host"
-      COMPONENT compiler
-    )
+    if(CMAKE_SYSTEM_NAME MATCHES Windows)
+      add_custom_command(OUTPUT ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${sharedlib}
+                         DEPENDS "${SWIFT_PATH_TO_EARLYSWIFTSYNTAX_BUILD_DIR}/bin/${sharedlib}"
+                         COMMAND ${CMAKE_COMMAND} -E copy_if_different ${SWIFT_PATH_TO_EARLYSWIFTSYNTAX_BUILD_DIR}/bin/${sharedlib} ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${sharedlib})
+      add_custom_target(copy_swiftSyntaxLibrary_${sharedlib}
+                        DEPENDS ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${sharedlib}
+                        COMMENT "copying ${sharedlib}")
+      swift_install_in_component(PROGRAMS ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${sharedlib}
+                                 DESTINATION bin
+                                 COMPONENT compiler)
+    else()
+      add_custom_command(OUTPUT "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}"
+                         DEPENDS "${SWIFT_SYNTAX_LIBRARIES_BUILD_DIR}/${sharedlib}"
+                         COMMAND ${CMAKE_COMMAND} -E copy_if_different ${SWIFT_SYNTAX_LIBRARIES_BUILD_DIR}/${sharedlib} ${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}
+                         ${add_origin_rpath})
+      add_custom_target(copy_swiftSyntaxLibrary_${sharedlib}
+                        DEPENDS "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}"
+                        COMMENT "Copying ${sharedlib}")
+      swift_install_in_component(PROGRAMS "${SWIFT_HOST_LIBRARIES_DEST_DIR}/${sharedlib}"
+                                 DESTINATION "lib${LLVM_LIBDIR_SUFFIX}/swift/host"
+                                 COMPONENT compiler)
+    endif()
 
     add_dependencies(swiftSyntaxLibraries copy_swiftSyntaxLibrary_${sharedlib})
   endforeach()
 
+  foreach(implib ${SWIFT_SYNTAX_IMPORT_LIBRARIES})
+    set(path "${SWIFT_PATH_TO_EARLYSWIFTSYNTAX_BUILD_DIR}/lib/swift/host/${implib}")
+    cmake_path(NATIVE_PATH path NORMALIZE source)
+
+    set(path "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/lib/swift/windows/${implib}")
+    cmake_path(NATIVE_PATH path NORMALIZE destination)
+
+    add_custom_command(OUTPUT ${destination}
+      DEPENDS ${source}
+      COMMAND "${CMAKE_COMMAND}" -E copy_if_different ${source} ${destination})
+    add_custom_target(copy_swiftSyntaxLibrary_${implib}
+      DEPENDS ${destination}
+      COMMENT "Copying ${implib}")
+    swift_install_in_component(PROGRAMS ${destination}
```

Comment:
IMO there's no reason to install these.

Reply:
Should we remove it for the other targets as well?  I'd rather keep the image in sync.  The installer is the authority on what is distributed.

Reply:
Which other targets? You can leave for now if you want, it doesn't hurt anything. We can remove all interfaces + tbd/importlib in main after all this anyway.

Reply:
macOS/Linux/BSD.